enhancements/oc/icsp-registry-scoped.md

```diff
@@ -0,0 +1,238 @@
+---
+title: generate-ImageContentSourcePolicy-scoped-to-a-registry
+authors:
+  - "@mhrivnak"
+  - "@sacharya"
+reviewers:
+  - "@mgoldboi"
+  - "@robszumski"
+  - "@beekhof"
+  - "@ecordell"
+approvers:
+  - TBD
+creation-date: 2020-06-10
+last-updated: 2020-06-10
+status: implementable
+---
+
+# Generate ImageContentSourcePolicy scoped to a registry
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+
+## Summary
+
+The `oc adm catalog mirror` command [generates an
+ImageContentSourcePolicy](https://docs.openshift.com/container-platform/4.4/operators/olm-restricted-networks.html#olm-updating-operator-catalog-image_olm-restricted-networks)
+that maps the original container image repository to a new location where it
+will be mirrored, typically inside a disconnected environment. When that ICSP
+is applied to a cluster, it is converted to a config file for cri-o and placed
+onto each Node. The process of placing the config file on a Node includes
+rebooting that Node.
+
+Today, the `oc adm catalog mirror` command generates an ICSP where each entry
+is specific to a repository. For example, it would map
+`registry.somevendor.io/cloud/burrito-db` to
+`mirror.internal.customer.com/cloud/burrito-db`. This enhancement proposes to
+start generating ICSPs only at the registry scope. Using the same example,
```

Comment:
```suggestion
optionally generate ICSPs at the registry scope. Using the same example,
```

I suspect we'll get further if this isn't the only, or default, way ICSPs are generated.

Reply:
Would this be the default behavior, and we'd use a command-line flag to generate the old style? I guess that would give us more ability to accommodate an unanticipated use case where someone wants to keep the current behavior.

I'm just not sure if or why anyone might be tied to the current behavior. Nobody on the long email thread chimed in to explain the current behavior vs the proposed wide approach.

Reply:
My guess is that they will want to preserve the old behaviour as the default.
But that is not based on any facts.

Reply:
Customers will have a better experience if the wide ICSPs are the default, so I'm hopeful we can accomplish that. I'll add a note suggesting a command-line flag to make it work the way it does today in case that turns out to be desirable, and we can see how that flies.